src/sources/soundsourceflac.cpp

```diff
@@ -345,25 +344,31 @@ FLAC__bool SoundSourceFLAC::flacEOF() {
     return m_file.atEnd();
 }
 
+namespace {
+
+// Workaround for improperly encoded FLAC files that may contain
+// garbage in the most significant, unused bits of decoded samples.
+// Required at least for libFLAC 1.3.2. This workaround might become
+// obsolete once libFLAC is taking care of these issues internally.
+// https://bugs.launchpad.net/mixxx/+bug/1769717
+// https://hydrogenaud.io/index.php/topic,61792.msg559045.html#msg559045
+//
+// We will shift decoded samples left by (32 - m_bitsPerSample) to
+// get rid of the garbage in the most significant bits before scaling.
+// The range of decoded integer sample values then becomes
+// [-2 ^ 31, 2 ^ 31 - 1]. Afterwards this integer range needs to be
+// scaled to [-CSAMPLE_PEAK, CSAMPLE_PEAK).
```

Comment:
[-2 ^ 31 - X, 2 ^ 31 - 1 - X] where X is the resolution  
scaled to [-CSAMPLE_PEAK, CSAMPLE_PEAK - X/2^31]
I would just remove the two last sentences, currently they are incorrect and later they will be confusing.

Reply:
How about this: 

// We will shift decoded samples left by (32 - m_bitsPerSample) to
// get rid of the garbage in the most significant bits before scaling.
// The resulting full scale 32 bit signed sample is divided by 2 ^ 31 to 
// normalize it back to a floating point < CSAMPLE_PEAK.